pgjdbc/src/main/java/org/postgresql/jdbc/EscapedFunctions2.java

```diff
@@ -0,0 +1,665 @@
+        || appendSingleIntervalCast(buf, SQL_TSI_HOUR, type, value, "hour")
+        || appendSingleIntervalCast(buf, SQL_TSI_MINUTE, type, value, "minute")
+        || appendSingleIntervalCast(buf, SQL_TSI_MONTH, type, value, "month")
+        || appendSingleIntervalCast(buf, SQL_TSI_WEEK, type, value, "week")
+        || appendSingleIntervalCast(buf, SQL_TSI_YEAR, type, value, "year")
+    ) {
+      return;
+    }
+    if (areSameTsi(SQL_TSI_QUARTER, type)) {
+      buf.append("CAST((").append(value).append("::int * 3) || ' month' as interval)");
+      return;
+    }
+    throw new PSQLException(GT.tr("Interval {0} not yet implemented", type),
+        PSQLState.NOT_IMPLEMENTED);
+  }
+
+  private static boolean appendSingleIntervalCast(StringBuilder buf, String cmp, String type, String value, String pgType) {
+    if (!areSameTsi(type, cmp)) {
+      return false;
+    }
+    buf.ensureCapacity(buf.length() + 5 + 4 + 14 + value.length() + pgType.length());
+    buf.append("CAST(").append(value).append("||' ").append(pgType).append("' as interval)");
+    return true;
+  }
+
+  /**
+   * Compares two TSI intervals. It is
+   * @param a first interval to compare
+   * @param b second interval to compare
+   * @return true when both intervals are equal (case insensitive)
+   */
+  private static boolean areSameTsi(String a, String b) {
+    return a.length() == b.length() && b.length() > SQL_TSI_ROOT.length()
+        && a.regionMatches(true, SQL_TSI_ROOT.length(), b, SQL_TSI_ROOT.length(), SQL_TSI_ROOT.length() - b.length());
+  }
+
+  /**
+   * Checks if given input starts with {@link #SQL_TSI_ROOT}
+   * @param interval input string
+   * @return true if interval.startsWithIgnoreCase(SQL_TSI_ROOT)
+   */
+  private static boolean isTsi(String interval) {
+    return interval.regionMatches(true, 0, SQL_TSI_ROOT, 0, SQL_TSI_ROOT.length());
+  }
+
+
+  /**
+   * time stamp diff
+   *
+   * @param parsedArgs arguments
+   * @throws SQLException if something wrong happens
+   */
+  public static void sqltimestampdiff(StringBuilder buf, List<? extends CharSequence> parsedArgs) throws SQLException {
+    if (parsedArgs.size() != 3) {
+      throw new PSQLException(
+          GT.tr("{0} function takes three and only three arguments.", "timestampdiff"),
+          PSQLState.SYNTAX_ERROR);
+    }
+    buf.append("extract( ")
+        .append(constantToDatePart(buf, parsedArgs.get(0).toString()))
+        .append(" from (")
+        .append(parsedArgs.get(2))
+        .append("-")
+        .append(parsedArgs.get(1))
+        .append("))");
+  }
+
+  private static String constantToDatePart(StringBuilder buf, String type) throws SQLException {
+    if (!isTsi(type)) {
+      throw new PSQLException(GT.tr("Interval {0} not yet implemented", type),
+          PSQLState.SYNTAX_ERROR);
+    }
+    if (areSameTsi(SQL_TSI_DAY, type)) {
+      return "day";
+    } else if (areSameTsi(SQL_TSI_SECOND, type)) {
```

Comment:
it is a style issue, but since all of these blocks return, they do not need to be in else if blocks

Reply:
Ah, both variants are unpleasant to me. I just kept it the same way it was written initially.
A slight plus is `else if` is less number of lines.